internal/controller/backup_controller.go

```diff
@@ -287,7 +287,10 @@ func (r *BackupReconciler) isValidBackupRunning(
 		return false, fmt.Errorf("unknown.spec.target received: %s", backup.Spec.Target)
 	}
 
-	containerIsNotRestarted := backup.Status.InstanceID.ContainerID == pod.Status.ContainerStatuses[0].ContainerID
+	var containerIsNotRestarted bool
+	if len(pod.Status.ContainerStatuses) > 0 {
+		containerIsNotRestarted = backup.Status.InstanceID.ContainerID == pod.Status.ContainerStatuses[0].ContainerID
```

Comment:
Is it possible that `backup.Status.InstanceID` is nil here? I think it could be after the change to the `SetAsStarted` function in this patch.

Reply:
Adding a further check makes a lot of sense.